### PR TITLE
SearchFilter: fix the click event on SearchFilter.Item not working

### LINKED ISSUE #380: SearchFilter: the click event on SearchFilter.Item is not working
Because `SearchFilter` is listening `mouseDown` event to close the menu, `SearchFilter.Item` is unmounted already before the click event on it is triggered.

### PR DESCRIPTION
**Pull request by rosslo**
Issue: https://github.com/glints-dev/glints-aries/issues/380

### CODE REVIEW

File: src/Input/SearchFilter/SearchFilter.tsx [hunk 1]
```diff
@@ -43,24 +53,6 @@ class SearchFilter extends React.Component<Props, State> {
     this.setState({ isOpen: false });
   };
 
-  handleMouseDown = (event: MouseEvent) => {
-    const element = event.target as HTMLElement;
-    const hasClickedOnInput = element === this.inputRef.current;
-    const hasClickedInsideSearchFilter = this.searchFilterRef.current.contains(
-      element
-    );
-    const hasClickedOnScrollBar =
```
Comment (tingtingdisco) [inline src/Input/SearchFilter/SearchFilter.tsx:52]: hasClickedOnScrollBar is not relevant anymore?

Comment (rosslo) [reply src/Input/SearchFilter/SearchFilter.tsx:52]: I think so. I tried to click the dropdown's scrollbar, and it doesn't close the menu.

File: src/Input/SearchFilter/SearchFilterItem.tsx [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
 const SearchFilterItem: React.FunctionComponent<Props> = props => {
   const { children, className, onClick, ...defaultProps } = props;
+  const { closeSearchFilterMenu } = useSearchFilterContext();
 
+  const handleClick = (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
+    if (onClick) {
+      onClick(e);
+    }
+    if (closeSearchFilterMenu) {
+      closeSearchFilterMenu();
```
Comment (tingtingdisco) [inline src/Input/SearchFilter/SearchFilterItem.tsx:14]: how the filter be closed before? why need to `closeSearchFilterMenu` here now?

File: src/Input/SearchFilter/Context.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { createContext, useContext } from 'react';
+
+interface SearchFilterContext {
+  closeSearchFilterMenu: () => void;
+}
+
+export const SearchFilterContext = createContext<
+  SearchFilterContext | undefined
```
Comment (tingtingdisco) [inline src/Input/SearchFilter/Context.ts:8]: could you explain why the type needs to be `SearchFilterContext | undefined`

**Review by tingtingdisco — APPROVED** [review]
LGTM

**Pull request merged**